### PR TITLE
Add support for India Residency

### PR DESCRIPTION
**Pull request by myronkaifung**
- Changed `eu` param to `residency`
- `Residency` is a str that accepts 'us', 'eu', or 'in'. Default value = "us"
- Updated readme to reflect the above

### CODE REVIEW

**Review by austinpray-mixpanel — APPROVED** [review]

**Review by austinpray-mixpanel — CHANGES_REQUESTED** [review]
Future proofing discussion happening in slack. Defer to @dongjae93

File: src/mixpanel_utils/__init__.py
Comment (dongjae93) [inline src/mixpanel_utils/__init__.py:115]: let's follow what Ted suggested and make this future proof. The validation logic ensures we will have valid residency value so we can do something like 
if us:  "https://data.mixpanel.com/api"
else: f"https://data-{residency}.mixpanel.com/api"

File: src/mixpanel_utils/__init__.py
Comment (dongjae93) [inline src/mixpanel_utils/__init__.py:120]: same as above

File: src/mixpanel_utils/__init__.py
Comment (dongjae93) [inline src/mixpanel_utils/__init__.py:125]: same as above

**Review by dongjae93 — CHANGES_REQUESTED** [review]

**Comment by myronkaifung** [comment]
@dongjae93 would you please review again?

**Review by dongjae93 — APPROVED** [review]
LGTM!

**Comment by myronkaifung** [comment]
Can you merge for me? I have no write access

**Pull request merged**
